snuba/consumers/consumer.py

```diff
@@ -334,39 +274,7 @@ def __build_write_step(self) -> ProcessedMessageBatchWriter:
             insert_batch_writer, replacement_batch_writer
         )
 
-    def create(
-        self, commit: Callable[[Mapping[Partition, int]], None]
-    ) -> ProcessingStrategy[KafkaPayload]:
-        collect = CollectStep(
-            self.__build_write_step,
-            commit,
-            self.__max_batch_size,
-            self.__max_batch_time,
-        )
-
-        transform_function = functools.partial(process_message, self.__processor)
-
-        strategy: ProcessingStrategy[KafkaPayload]
-        if self.__processes is None:
-            strategy = TransformStep(transform_function, collect)
-        else:
-            assert self.__input_block_size is not None
-            assert self.__output_block_size is not None
-            strategy = ParallelTransformStep(
-                transform_function,
-                collect,
-                self.__processes,
-                max_batch_size=self.__max_batch_size,
-                max_batch_time=self.__max_batch_time,
-                input_block_size=self.__input_block_size,
-                output_block_size=self.__output_block_size,
-                metrics=StreamMetricsAdapter(MetricsWrapper(self.__metrics, "process")),
-            )
-
-        if self.__prefilter is not None:
-            strategy = FilterStep(self.__should_accept, strategy)
-
-        return strategy
+    return build_writer
```

Comment:
Who is actually using this function ?

Reply:
This is the writer function passed to the collect step and called on every collect. It should be fundamentally the same as before where it lived in the streaming strategy factory https://github.com/getsentry/snuba/blob/master/snuba/consumers/consumer.py#L318-L335

Reply:
Ah damn, I messed up indentation. I thought this was the return of `build_writer` itself, which would make no sense. Makes sense then.